### ISSUE TITLE
"minikube like" kismatic virtual machine installation for training

### ISSUE THREAD

**Issue by codymccain** [apprenda/kismatic#191]
# Problem Statement
I conduct introductory Kubernetes training on a regular basis and often have my students install a local [minikube](https://github.com/kubernetes/minikube) instance on their local workstation for hands-on exercises.  I would prefer to use a "minikube" like installation of Kismatic for this instead as it would ensure parity between training and production environments and better prepare the students for using specific Kismatic integrations.

A few features that minikube provides out of the box include that I would like to see the Kismatic CLI implement include:

## Provisioning to VirtualBox and other VMs
minikube has the capability to launch a local single host kubernetes cluster in a VM in a single command.  I am currently extending kismatic-provision to provide this (see https://github.com/apprenda/kismatic-provision/issues/9)

## Docker Machine / Repo integration
see https://github.com/apprenda/kismatic/issues/193

with minikube, one can type
```
$ minikube docker-env
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.99.101:2376"
export DOCKER_CERT_PATH="/Users/marshallmccain/.minikube/certs"
export DOCKER_API_VERSION="1.23"
# Run this command to configure your shell:
# eval $(minikube docker-env)
```

which provides an easy way for the user to configure their docker client to use the minikube docker daemon and repo for building and deploying containers.  Having this would remove any need to have a docker repo available when deploying workloads into the repository.  Kismatic currently provides an easy configuration parameter in the plan file to install a local repo, however, additional configurations have to be performed to use it and docker images must be explicitly pushed to it.  

Having the above command would simplify the user's experience when connecting their local docker cli to the daemon and repo (see https://docs.docker.com/registry/insecure/ for the additional configuration that is currently required)

## SSH Integration
see https://github.com/apprenda/kismatic/issues/194

When a user sets up a Kismatic installation, they are required to provide a public/private ssh key pair  for accessing the target hosts.  If a minikube like cluster is automatically provisioned and launched, it would be useful to allow the user to SSH into the VM host by issuing:

```
$ kismatic ssh <?nodename>
```

Obviously I am only addressing a single host machine installation with this syntax.  Maybe we default to the first node listed in the plan file if a node name is not provided.

## Implement shortcut to dashboard
see https://github.com/apprenda/kismatic/issues/195

Minikube also provides the ability to quickly navigate a user to the Kubernetes Dashboard from the command line.  We should implement:

```
$ kismatic dashboard
```

to provide similar functionality.

## Implement command to extract the IP of a host
see https://github.com/apprenda/kismatic/issues/196

Minikube provides the  `minikube ip` command to access the IP address of a minikube instance.  Since small vm demo environments can be setup without DNS, this is useful in preparing lab instructions as the student can use this instead of substituting an IP address into a command string.  For example, if an http service has an ingress at `/foobar`, a student can type:

```
curl $(kismatic ip)/foobar
```

if using `BASH` and `curl` to access a REST endpoint for example.

**Comment by alexbrand** [comment]
Thanks @codymccain! May I suggest breaking this up into multiple issues? It seems like there are a lot of good ideas in here that are hard to parse as a single issue.

**Comment by codymccain** [comment]
@alexbrand issues have been broken out and referenced above in initial comment

**Comment by dasmb** [comment]
Closing; will be tracked on the broken-out issues.
